pyqtgraph/functions.py

```diff
@@ -1719,28 +1722,23 @@ def arrayToQPath(x, y, connect='all'):
     ## see: https://github.com/qt/qtbase/blob/dev/src/gui/painting/qpainterpath.cpp
 
     ## All values are big endian--pack using struct.pack('>d') or struct.pack('>i')
-
     path = QtGui.QPainterPath()
-
     n = x.shape[0]
 
-    # create empty array, pad with extra space on either end
```

Comment:
I think the arr / backstore construction is now used only if we fall through to the 'finite' connectivity case. It might help performance in the other paths to move the code below this to only run in the 'finite' branch.

Reply:
yeah, `finite` and `array` are the two use-cases remaining... I think there is a good chance we can boost performance on those methods too; in which case we can scrap a huge chunk of it.  if it's not looking like that is feasible before merging; I will restructure accordingly.